### PR TITLE
[DOC] Add MirrorMaker property descriptions to appendix

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Moved the property descriptions from the Kafka MirrorMaker configuration and  Kafka MirrorMaker 2.0 configuration sections
MM and MM2 procedure callouts link to the appropriate sections
Not every property is described for MM in the appendix, just covering the properties that were described in the main part of the guide.

Part of a work in progress to move all property descriptions into one place.

- Removed 14 files
- Additional property description content added through documentation/api/*.adoc files, which are included and built with the appendix
- New common properties section for all components: modules/con-common-configuration-proprieties.adoc
- Removed the :context: settings defined in the mirrormaker config assemblies, as they're no longer needed - no replication.
- Moved the full descriptions of JVM options and Container Images into the appendix. Non-MM components retain a small section that link to these, until they have procedure callouts to link from. 
- Moved the Kafka Connect logging section into appendix (for MM2), again retaining a small section.
- Made formats consistent for api adoc files (.title to describe a property)

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/con-common-configuration-properties.adoc [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+The bootstrap server lists can refer to Kafka clusters that are not deployed in the same Kubernetes cluster.
+They can also refer to a Kafka cluster not deployed by Strimzi.
+
+If on the same Kubernetes cluster, each list must ideally contain the Kafka cluster bootstrap service which is named `_CLUSTER-NAME_-kafka-bootstrap` and a port of 9092 for plain traffic or 9093 for encrypted traffic.
+If deployed by Strimzi but on different Kubernetes clusters, the list content depends on the approach used for exposing the clusters (routes, nodeports or loadbalancers).
+
+When using Kafka with a Kafka cluster not managed by Strimzi, you can specify the bootstrap servers list according to the configuration of the given cluster.
+
+[id='con-common-configuration-cipher-{context}']
+.`SSL` cipher suites
```
Comment (scholzj) [inline documentation/modules/con-common-configuration-properties.adoc:40]: This does not describe only cipher suite configuration. So maybe it should be named _SSL configuration`?

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
-<2> The SSl protocol `TLSv1.2` is enabled.
-<3> Specifies the `TLSv1.2` protocol to generate the SSL context.
-Allowed values are `TLSv1.1` and `TLSv1.2`.
+[id='property-mirrormaker2-config-{context}']
+.`config`
+Use the `config` properties to configure Kafka options.
+
+Standard Apache Kafka configuration may be provided, xref:assembly-kafka-connect-configuration-deployment-configuration-kafka-connect[restricted to those properties not managed directly by Strimzi].
+
+To run external listeners with a specific _cipher suite_ for a TLS version, you can xref:con-common-configuration-cipher-reference[configure allowed `ssl` properties].
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc:9]: This does not configure listeners but clients.

File: documentation/modules/con-common-configuration-properties.adoc [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+
+.Example SSL configuration
+[source,yaml,subs="attributes+"]
+----
+spec:
+  producer:
+    config:
+      ssl.cipher.suites: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384" <1>
+      ssl.enabled.protocols: "TLSv1.2" <2>
+      ssl.protocol: "TLSv1.2" <3>
```
Comment (scholzj) [inline documentation/modules/con-common-configuration-properties.adoc:53]: Thsi shows up in many different variants ... I wonder if it should be just

```suggestion
# ...
config:
  ssl.cipher.suites: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384" <1>
  ssl.enabled.protocols: "TLSv1.2" <2>
  ssl.protocol: "TLSv1.2" <3>
# ...
```

in the example.

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerConsumerSpec.adoc [hunk 4]
```diff
@@ -1,18 +1,65 @@
...
+
+*Exceptions*
+
+You can specify and configure the options listed in the {ApacheKafkaConsumerConfig}.
+
+However, there are exceptions for options automatically configured and managed directly by Strimzi related to:
+
+* Kafka cluster bootstrap address
+* Security (encryption, authentication, and authorization)
+* Consumer group identifier
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerConsumerSpec.adoc:41]: We should add _Interceptor classes_ to the list

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerConsumerSpec.adoc [hunk 5]
```diff
@@ -1,18 +1,65 @@
...
+
+Specifically, all configuration options with keys equal to or starting with one of the following strings are forbidden:
+
+* `ssl.` (with the exception of xref:con-common-configuration-cipher-reference[cipher suites])
+* `sasl.`
+* `security.`
+* `bootstrap.servers`
+* `group.id`
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerConsumerSpec.adoc:49]: We should add `interceptor.classes` to the list

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerProducerSpec.adoc [hunk 6]
```diff
@@ -1,18 +1,50 @@
...
+* Kafka cluster bootstrap address
+* Security (encryption, authentication, and authorization)
+
+Specifically, all configuration options with keys equal to or starting with one of the following strings are forbidden:
+
+* `ssl.` (with the exception of xref:con-common-configuration-cipher-reference[cipher suites])
+* `sasl.`
+* `security.`
+* `bootstrap.servers`
+* `group.id`
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerProducerSpec.adoc:43]: We should add `interceptor.classes` to the list and remove `group.id` which is consumer only.

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerProducerSpec.adoc [hunk 7]
```diff
@@ -1,18 +1,50 @@
...
+To run external listeners with a specific _cipher suite_ for a TLS version, you can xref:con-common-configuration-cipher-reference[configure allowed `ssl` properties].
+
+*Exceptions*
+
+You can specify and configure the options listed in the {ApacheKafkaProducerConfig}.
+
+However, there are exceptions for options automatically configured and managed directly by Strimzi related to:
+
+* Kafka cluster bootstrap address
+* Security (encryption, authentication, and authorization)
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerProducerSpec.adoc:35]: We should add Interceptor classes to the list

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerTls.adoc [hunk 8]
```diff
@@ -1,5 +1,8 @@
-Use the `tls` property to configure TLS encryption.
-Provide a list of secrets with key names under which the certificates are stored in X.509 format.
+Use the `KafkaMirrorMakerTls` properties to configure TLS trusted certificates for connecting MirrorMaker to the cluster.
+
+.`trustedCertificates`
+Having xref:type-KafkaClientAuthenticationTls-reference[set `tls` to configure TLS encryption], use the
+`trustedCertificates` property to provide a list of secrets with key names under which the certificates are stored in X.509 format.
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerTls.adoc:5]: Can we add something like this at the end?

```
If you want to enable TLS but use the default set of public certification authorities shipped with Java, you can specify the `trustedCertificates` as empty array:

.An example of enabling TLS with the default Java certificates 
[source,yaml,subs=attributes+]
----
tls:
  trustedCertificates: []
----
```

File: documentation/api/io.strimzi.api.kafka.model.authentication.KafkaClientAuthenticationOAuth.adoc [hunk 9]
```diff
@@ -44,11 +42,11 @@ authentication:
 ----
 
 .Access token
-
 You can configure the access token used for authentication with the Kafka broker directly.
 In this case, you do not specify the `tokenEndpointUri`.
 In the `accessToken` property, specify a link to a `Secret` containing the access token.
 
+
```
Comment (scholzj) [inline documentation/api/io.strimzi.api.kafka.model.authentication.KafkaClientAuthenticationOAuth.adoc:49]: Do we need the 2 empty lines here?

File: documentation/modules/con-common-configuration-properties.adoc [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+The type of replication depends on the resource.
+
+* `KafkaTopic` uses a replication factor to configure the number of replicas of each partition within a Kafka cluster.
+* Kafka components use replicas to configure the number of pods in a deployment to provide better availability and scalability.
+
+*MirrorMaker*
+
+When running Kafka MirrorMaker on Kubernetes it is not absolutely necessary to run multiple replicas of the Kafka MirrorMaker for high availability.
+When the node where the Kafka MirrorMaker has deployed crashes, Kubernetes will automatically reschedule the Kafka MirrorMaker pod to a different node.
+However, running Kafka MirrorMaker with multiple replicas can provide faster failover times as the other nodes will be up and running.
```
Comment (scholzj) [inline documentation/modules/con-common-configuration-properties.adoc:24]: This applies to also for example Kafka Connect. But it seems a bit weird to mention it explicitely in section called _Common configuration properties_.

File: documentation/api/io.strimzi.api.kafka.model.KafkaAuthorizationOpa.adoc [hunk 11]
```diff
@@ -1,30 +1,30 @@
-To use link:https://www.openpolicyagent.org/[Open Policy Agent^] authorization, set the `type` property in the `authorization` section to the value `opa`.
-The Open Policy Agent authorizer has several configuration options:
+To use link:https://www.openpolicyagent.org/[Open Policy Agent^] authorization, set the `type` property in the `authorization` section to the value `opa`,
```
Comment (ppatierno) [inline documentation/api/io.strimzi.api.kafka.model.KafkaAuthorizationOpa.adoc:1]: as far as I understood these changes on OPA related stuff are not related to the PR purpose. Can we do that in a different PR?

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaAuthorizationOpa.adoc:1]: They're just a few formatting changes to make consistent with the other descriptions added to the appendix. So it's sort of related.

File: documentation/modules/con-common-configuration-properties.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:40]: It's all _configuration_. Trying to use the property names in the title, so maybe .`ssl` is enough.

Comment (scholzj) [reply documentation/modules/con-common-configuration-properties.adoc:40]: Then it should either just SSL or maybe _SSL protocol versions and cipher suites_

Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc]: I've left at _ssl_

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc]: I changed to _For client connection using a specific cipher suite_ and assumed we needed this change wherever else this sentence was used.

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerTls.adoc (hunk 8 above)
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerTls.adoc:5]: Added

**Comment by scholzj** [comment]
@PaulRMellor Could you please look at the conflicts:

**Comment by PaulRMellor** [comment]
@scholzj After the merge conflict fix, I fixed a few broken links for the template updates. 
I also changed the format of the template KafkaContainer config, as it wasn't presenting correctly

16f0781caee677a0181184b482eb2f9b255e6c70
